accounts/addproofer.php

```diff
@@ -77,172 +246,29 @@ function _validate_fields($real_name, $username, $userpass, $userpass2, $email,
         return _("That user name already exists, please try another.");
     }
 
+    // Ensure we don't already have a registration with this name
+    try
+    {
+        $na_user = new NonactivatedUser($username);
+        return _("That username has already been requested. Please try another.");
+    }
+    catch(NonexistentNonactivatedUserException$exception)
+    {
+        // pass
+    }
+
     // TODO: The above check only validates against users in the DP database.
     // It's possible that there are usernames already registered with the
     // underlying forum software (like 'Anonymous') or are disallowed in the
     // forum software which, if used, will cause account creation to fail in
     // activate.php.
 
-    return '';
-}
-
-// ---------------------------------------------------------------------------
-
-// assume there is no error
-$error = "";
-
-$password = isset($_POST['password'])? $_POST['password']: '';
-if ($password=="proofer") {
-
-    // From the form filled out at the end of this file
-
-    $real_name = $_POST['real_name'];
-    $username = $_POST['userNM'];
-    $userpass = $_POST['userPW'];
-    $email = @$_POST['email'];
-    $userpass2 = $_POST['userPW2'];
-    $email2 = @$_POST['email2'];
-    $email_updates = $_POST['email_updates'];
-
-    // When in testing mode, to avoid leaking private email addresses,
-    // create a fake but distinct email address based on the username.
-    // DP usernames allow [0-9A-Za-z@._ -]. '@' and ' ' are not valid
-    // (unless quoted) in the local part of an email address, so
-    // convert those to '%' and '+' respectively.
-    if ($testing) {
-        $local_part = str_replace(array('@', ' '), array('%', '+'), $username);
-        $email      = $local_part . "@localhost";
-        $email2     = $email;
-    }
-
-    // Run all form validators against the data
-    foreach($form_validators as $func)
+    if(!$referrer)
     {
-        $error = $func($real_name, $username, $userpass, $userpass2, $email, $email2, $email_updates);
-        if(!empty($error))
-            break;
+        return _("Please tell us how you heard from us.");
```

Comment:
"from"? Presumably they haven't heard anything *from* us yet. Change to "about"?